### PR TITLE
Make all drawing call asynchronous

### PR DESCRIPTION
**Pull request by xjuan**
This improves responsiveness a lot.

### CODE REVIEW

**Comment by tchx84** [comment]
I tested this on a real app with flapjack, and the difference is quite noticeable, e.g. no more visual artifacts and lag when resizing the screen.

File: src/maxwell-web-view.c [hunk 1]
```diff
@@ -40,10 +40,17 @@ typedef struct
...
+  guint      id;              /* Unique id */
+  GdkPixbuf *pixbuf;          /* A Pixbuf */
+} PixbufData;
+
 typedef struct
 {
   GList        *children;     /* List of ChildData */
-  GList        *pixbufs;      /* List of temp pixbufs to handle maxwell:// requests */
+  GQueue       *pixbufs;      /* List of PixbufData to handle maxwell:// requests */
+  guint         pixbuf_count; /* Pixbuf counter, used ad id */
```
Comment (ptomato) [inline src/maxwell-web-view.c]: typo: ad -> as

File: src/maxwell-web-view.js [hunk 2]
```diff
@@ -195,13 +183,34 @@ window.maxwell.child_resize = function (id, width, height, minWidth, minHeight)
...
+
+        try {
+            let image = new ImageData(data, dReq.maxwell.width, dReq.maxwell.height);
+            /* Update contents */
+            ctx.putImageData(image, dReq.maxwell.x, dReq.maxwell.y);
+        } catch (error) {
+        }
     }
+
+    /* Remove all request that where drawn */
```
Comment (ptomato) [inline src/maxwell-web-view.js]: typo: where -> were

File: src/maxwell-web-view.js [hunk 3]
```diff
@@ -219,13 +228,19 @@ window.maxwell.child_draw = function (id, image_id, x, y, width, height) {
...
 
-    /* Update contents */
-    if (image) {
-        let ctx = child.getContext('2d');
-        ctx.globalCompositeOperation = "copy";
-        ctx.putImageData(image, x, y);
+    xhr.open('GET', 'maxwell:///' + image_id);
+    xhr.responseType = 'arraybuffer';
+    xhr.addEventListener('load', on_child_draw_load);
+    xhr.maxwell = { child: child, x: x, y: y, width: width, height: height };
```
Comment (ptomato) [inline src/maxwell-web-view.js]: optional: If we have a new enough JavaScriptCore you may be able to use the nice ES6 syntax: `xhr.maxwell = {child, x, y, width, height};`

**Review by ptomato — CHANGES_REQUESTED** [review]
The code looks fine, but now when I run it with flapjack the Maxwell canvas is entirely blank.

**Comment by tchx84** [comment]
Tested this again, and I can't reproduce the issue @ptomato mentioned. It looks OK here. BUT, I noticed some "on_maxwell_uri_scheme_request /%d" message trashing the log.

**Review by ptomato — APPROVED** [review]
We discussed that we'll merge this, but still continue to investigate what the issue is with my machine.

**Pull request merged**